airflow-core/src/airflow/serialization/serializers/pydantic.py

```diff
@@ -0,0 +1,78 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import annotations
+
+from importlib import import_module
+from typing import TYPE_CHECKING, cast
+
+from pydantic import BaseModel
+
+from airflow.serialization.serde import _is_pydantic_basemodel
+from airflow.utils.module_loading import qualname
+
+if TYPE_CHECKING:
+    from airflow.serialization.serde import U
+
+serializers = [
+    "pydantic.main.BaseModel",
+]
+deserializers = serializers
+stringifiers = serializers
+
+__version__ = 1
+
+
+def _resolve_pydantic_class(qn: str):
+    module_name, class_name = qn.rsplit(".", 1)
+    module = import_module(module_name)
+    return getattr(module, class_name)
+
+
+def serialize(o: object) -> tuple[U, str, int, bool]:
+    if not _is_pydantic_basemodel(o):
+        return "", "", 0, False
+
+    # to convince mypy
+    m = cast("BaseModel", o)
+    # Serialize
+    data = m.model_dump()
+    # Store the actual qualified name for the pydantic class. This classname will be used to import the module and load the data.
+    data["__class__"] = qualname(o)
```

Comment:
We can just do:
```
f"{model.__class__.__module__}.{model.__class__.__qualname__}"
```

instead?

Reply:
That’s basically what `qualname` does. (It uses `__name__` instead of `__qualname__`, but they are the same in this case.)